command/certificate/remote.go

```diff
@@ -57,19 +64,23 @@ func getPeerCertificates(addr, serverName, roots string, insecure bool) ([]*x509
 // empty string (and 'isURL:false').
 //
 // Examples:
-// trimURL("https://smallstep.com/onbaording") -> "smallstep.com", true, nil
+// trimURL("https://smallstep.com/onbaording") -> "smallstep.com:443", true, nil
 // trimURL("https://ca.smallSTEP.com:8080") -> "ca.smallSTEP.com:8080", true, nil
 // trimURL("./certs/root_ca.crt") -> "", false, nil
-// trimURL("hTtPs://sMaLlStEp.cOm") -> "sMaLlStEp.cOm", true, nil
+// trimURL("hTtPs://sMaLlStEp.cOm") -> "sMaLlStEp.cOm:443", true, nil
 // trimURL("hTtPs://sMaLlStEp.cOm hello") -> "", false, err{"invalid url"}
 func trimURL(ref string) (string, bool, error) {
 	tmp := strings.ToLower(ref)
-	for _, prefix := range urlPrefixes {
+	for prefix, _ := range urlPrefixes {
 		if strings.HasPrefix(tmp, prefix) {
 			u, err := url.Parse(ref)
 			if err != nil {
 				return "", false, errors.Wrapf(err, "error parsing URL '%s'", ref)
 			}
+			if _, _, err := net.SplitHostPort(u.Host); err != nil {
+				port := uint64(urlPrefixes[prefix])
+				u.Host += ":" + strconv.FormatUint(port, 10)
```

Comment:
Could you use net.JoinHostPort here to handle IPv6

Reply:
Done.

https://github.com/smallstep/cli/pull/653/commits/46f908a925bcc292ae848908296eab61e099dc78